File: jsonRecycle/app/src/main/java/com/example/jsonrecycle/MainActivity.java
package com.example.jsonrecycle;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.Volley;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import org.json.JSONArray;
import java.lang.reflect.Type;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private HeroAdapter heroAdapter;
    private List<Hero> heroList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        fetchHeroes();
    }

    private void fetchHeroes() {
        String url = "https://simplifiedcoding.net/demos/marvel/";

        JsonArrayRequest request = new JsonArrayRequest(Request.Method.GET, url, null,
                new Response.Listener<JSONArray>() {
                    @Override
                    public void onResponse(JSONArray response) {
                        Type listType = new TypeToken<List<Hero>>(){}.getType();
                        heroList = new Gson().fromJson(response.toString(), listType);
                        heroAdapter = new HeroAdapter(MainActivity.this, heroList);
                        recyclerView.setAdapter(heroAdapter);
                    }
                },
                new Response.ErrorListener() {
                    @Override
                    public void onErrorResponse(VolleyError error) {
                        Toast.makeText(MainActivity.this, "Failed to fetch data", Toast.LENGTH_SHORT).show();
                    }
                });

        RequestQueue queue = Volley.newRequestQueue(this);
        queue.add(request);
    }
}

File: Recycler/app/src/main/java/com/example/recycler/MainActivity.kt
package com.example.recycler

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: ItemAdapter
    private lateinit var itemList: MutableList<Item>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)

        itemList = generateItems()
        adapter = ItemAdapter(itemList)
        recyclerView.adapter = adapter
    }

    private fun generateItems(): MutableList<Item> {
        val items = mutableListOf<Item>()
        items.add(Item(R.drawable.iphone, "Iphone"))
        items.add(Item(R.drawable.samsung, "Samsung"))
        items.add(Item(R.drawable.nokia, "Nokia"))
        items.add(Item(R.drawable.realme, "Realme"))
        items.add(Item(R.drawable.oppo, "OPPO"))
        return items
    }
}
